jbpm-services/jbpm-executor/src/main/java/org/jbpm/executor/commands/ExecutionErrorCleanupCommand.java

```diff
@@ -99,44 +86,7 @@ public ExecutionResults execute(CommandContext ctx) throws Exception {
 		// get hold of persistence and create instance of audit service
 		EntityManagerFactory emf = EntityManagerFactoryManager.get().getOrCreate(emfName);
 
-		// collect parameters
-		String olderThan = (String)ctx.getData("OlderThan");
-		String olderThanPeriod = (String)ctx.getData("OlderThanPeriod");
-		String forProcess = (String)ctx.getData("ForProcess");
-		String forProcessInstance = (String)ctx.getData("ForProcessInstance");
-		String forDeployment = (String)ctx.getData("ForDeployment");
-		
-		if (olderThanPeriod != null) {
-			long olderThanDuration = DateTimeUtils.parseDateAsDuration(olderThanPeriod);
-			Date olderThanDate = new Date(System.currentTimeMillis() - olderThanDuration);
-			
-			olderThan = formatToUse.format(olderThanDate);
-		}
-		Map<String, Object> parameters = new HashMap<>();
-        StringBuilder cleanUpErrorsQuery = new StringBuilder();
-        
-        cleanUpErrorsQuery.append("delete from ExecutionErrorInfo where processInstanceId in "
-                + "(select processInstanceId from ProcessInstanceLog where status in (2,3))");
-        if (olderThan != null && !olderThan.isEmpty()) {            
-            cleanUpErrorsQuery.append(" and errorDate < :olderThan");
-            parameters.put("olderThan", formatToUse.parse(olderThan));
-        }
-        if (forProcess != null && !forProcess.isEmpty()) {            
-            cleanUpErrorsQuery.append(" and processId = :forProcess");
-            parameters.put("forProcess", forProcess);
-        }
-        if (forProcessInstance != null && !forProcessInstance.isEmpty()) {            
-            cleanUpErrorsQuery.append(" and processInstanceId = :forProcessInstance");
-            parameters.put("forProcessInstance", Long.parseLong(forProcessInstance));
-        }
-        if (forDeployment != null && !forDeployment.isEmpty()) {            
-            cleanUpErrorsQuery.append(" and deploymentId = :forDeployment");
-            parameters.put("forDeployment", forDeployment);
-        }
-        
-        TransactionalCommandService commandService = new TransactionalCommandService(emf);
-        
-        int deletedErrors = commandService.execute(new UpdateStringCommand(cleanUpErrorsQuery.toString(), parameters));
+		int deletedErrors = new ExecutionErrorCleanup(emf).cleanup(ctx);
```

Comment:
Can we change this to static usage, I mean, use ExecutionErrorCleanupHelper.cleanup(emf, ctx)